packages/atomic/src/components/common/smart-snippets/atomic-smart-snippet-collapse-wrapper/atomic-smart-snippet-collapse-wrapper.pcss

```diff
@@ -27,7 +28,8 @@
 
 :host(.expanded) {
   &::part(smart-snippet-collapse-wrapper) {
-    height: var(--full-height);
+    height: auto;
+    max-height: calc(var(--full-height) + 200px);
```

Comment:
what happens when `max-height` isn't present here? Does it break the transition or does it still work?

AFAIK, setting a huge value like you said would break the transition, since it would lerp between the smaller value and the huge value, causing a delay before the animation starts showing visualy.

Also, would it be possible to remove this `+ 200px`? same reason as I mentioned above with the huge value.

Reply:
Hey Benjamin ! Did a little update that mostly makes the animation more consistent when collapsing the smart snippet. Also added a gif so that you can see it in action.
Tested with a static value of `--full-height` set to 1000 and it seemed to work really well as well.
I'm keeping the `+200px` as it alleviates cases where `.getBoundingClientRect()` doesn't give the correct size. 
Let me know what you think of the animation in the gif above !

Reply:
> mostly makes the animation more consistent when collapsing the smart snippet.

What was the inconsistency before?

> Tested with a static value of `--full-height` set to 1000 and it seemed to work really well as well.

Did you try with a huge value like you said, like `999999999px`? What happens?

> I'm keeping the `+200px` as it alleviates cases where `.getBoundingClientRect()` doesn't give the correct size.

What's the reason why it doesn't give the correct size? Are we certain its error will never exceed 200px? Otherwise, this will cut off the snippet and will most likely end up as a bug report.

Reply:
also, is there a delay between clicking to expand/collapse and the animation starting, if you wait like 2 seconds between pressing "show more", then "show less", then "collapse"?

Reply:
>What was the inconsistency before?

When the max height was larger, the animation was jumpy

> Did you try with a huge value like you said, like 999999999px? What happens?

I've tried with a value of `999999px` and I tweaked the transitions a bit to be 
```
:host(.expanded) {
  &::part(smart-snippet-collapse-wrapper) {
   /* ... */
    transition: max-height 2s cubic-bezier(0, 1, 0.22, 1) -1.75s;
  }
}

:host(.expanded) {
  &::part(smart-snippet-collapse-wrapper) {
    /* ... */
    transition: max-height 2s cubic-bezier(1, 0, 1, 0) 0.01s;
    /* ... */
  }
```
and I got the following animation 

https://github.com/coveo/ui-kit/assets/71142397/451959e8-cfc3-4c40-8cdf-dfa099265f2a

In my humble opinion this is probably good enough for 99% of cases ? I'll let you be the judge of that.

> What's the reason why it doesn't give the correct size? Are we certain its error will never exceed 200px? Otherwise, this will cut off the snippet and will most likely end up as a bug report.

As we discussed previously, the `getBoundingRect` acts up sometimes, I'm guessing that the inner slot is not playing nice with it and about half of the time I get a size that is slightly smaller that the actual height.

> also, is there a delay between clicking to expand/collapse and the animation starting, if you wait like 2 seconds between pressing "show more", then "show less", then "collapse"?

As you can see in my gif above, no delay ! 👍